aiida/manage/configuration/config.py

```diff
@@ -39,14 +38,22 @@ class Config(object):  # pylint: disable=too-many-public-methods
     def from_file(cls, filepath):
         """Instantiate a configuration object from the contents of a given file.
 
+        .. note:: if the filepath does not exist it will be created with the default configuration.
```

Comment:
Where in the function is the "default content" written to the file?

Reply:
It is not. The file is touched but is initially empty. The `Config` instance is then created with the default configuration but is not stored explicitly. This is done by the caller of `from_file` if needed. In our case this is done in `load_config` that calls `Config.from_file`. If I add to the docstring that the file is touched but the content is not written, is that clear enough?

Reply:
Taking a step back, would it not be cleaner if `.from_file` simply excepts if the filepath being passed does not exist?
If one wants to create a default configuration, shouldn't one simply all the constructor, followed by `.store()`?

I find this intermediate step confusing, where the configuration file is created but is empty.

Reply:
This design followed more or less directly from your other request (with which I agree) to not have warning messages being printed in the `Config` constructor. So the detection of whether the configuration has to be migrated as well as the actual migration (including the warning messages and backup generation) has to be done elsewhere. I think putting this in the `from_file` class method makes a lot of sense. Then the question was just how to deal with non-existing files. I think it is fine to put this here as well. This way one can always just call `from_file` also in all the unittest utils that don't have to worry about creating the file first or creating an initial default profile

Reply:
> Then the question was just how to deal with non-existing files. I think it is fine to put this here as well. This way one can always just call from_file also in all the unittest utils that don't have to worry about creating the file first or creating an initial default profile

I agree that it is a little bit less code to write, and so I'm not totally against it (but I would still say that this stretches the meaning of what a`.from_file` method is supposed to do).
Let's keep it for the moment - in that case, the question still remains whether this intermediate stage is necessary, where the configuration file is simply "touched" but without the actual content. 

If the configuration file does not exist, can we not simply write the default content to it?

Reply:
No you are right, the indirect is not necessary. I will make that final simplification.